identity/webapp/src/utility/logging.ts

```diff
@@ -0,0 +1,45 @@
+import { format as formatUrl, parse } from 'url'; // eslint-disable-line node/no-deprecated-api
+
+// This function redacts query parameters in URLs, so we have a URL that's
+// safe to log.  For example, we might get a log including:
+//
+//      /account/registration?session_token=eyJhbGciOi...{some sort of jwt}
+//
+// where the JWT could be decoded to reveal personal information including
+// email address and IP address.
+//
+// This function will replace the URL with:
+//
+//      /account/registration?session_token=[redacted]
+//
+// It is deliberately un-picky about what it redacts, because we'd rather
+// remove something innocent (e.g. ?refresh=true) than miss something sensitive.
+export const redactUrl = (url: string): string => {
+  // Note: we use a deprecated API here because we're working with
+  // relative URLs, e.g. `/account`.
+  //
+  // The WHATWG URL API that the deprecation message suggests doesn't
+  // work for this use case; it wants absolute URLs.
+  const parsedUrl = parse(url);
+
+  // If there are no query parameters to redact, we don't need to do anything.
+  if (parsedUrl.query === null) {
+    return url;
+  }
```

Comment:
This is the one place where this function differs from the code copied out of `app.ts`.